### PR TITLE
Add template with API Version 2017-01-31

### PR DESCRIPTION
**Pull request by raveeram**

### CODE REVIEW

**Review by SorraTheOrc — COMMENTED** [review]
Can you put this into a subdirectory (e.g. examples/descriptivename) and add a README that tells people how to use it (fine for the README to say "this template is... for details on how to use it see http://) - we can improve from there,

File: examples/Templates/azuredeploy.parameters.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "dnsNamePrefix": {
```
Comment (anhowe) [inline examples/Templates/azuredeploy.parameters.json]: replace with two: masterDnsNamePrefix, and agentDnsNamePrefix

File: examples/Templates/azuredeploy.parameters.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      "value": ""
+    },
+    "agentCount": {
+      "value": 3
+    },
+    "agentVMSize": {
+      "value": "Standard_D2_v2"
+    },
+    "adminUsername": {
```
Comment (anhowe) [inline examples/Templates/azuredeploy.parameters.json]: replace with linuxAdminUsername and windowsAgentAdminUsername

File: examples/Templates/azuredeploy.parameters.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      "value": ""
+    },
+    "windowsAdminPassword":{
```
Comment (anhowe) [inline examples/Templates/azuredeploy.parameters.json]: should be windowsAgentAdminPassword

Comment (raveeram) [reply examples/Templates/azuredeploy.parameters.json]: Ah, thanks. i haven't pushed an updated params file. Will do once I do some testing.

File: examples/Templates/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        "description": "The number of agents for the cluster.  This value can be from 1 to 100 (note, for Kubernetes clusters you will also get 1 or 2 public agents in addition to these seleted masters)"
+      },
+      "minValue":1,
+      "maxValue":100
+    },
+    "agentVMSize": {
+      "type": "string",
+      "defaultValue": "Standard_D2",
+      "allowedValues": [
+    		"Standard_A0", "Standard_A1", "Standard_A2", "Standard_A3", "Standard_A4", "Standard_A5",
```
Comment (anhowe) [inline examples/Templates/azuredeploy.json]: use values from here: https://github.com/Azure/acs-engine/blob/master/pkg/acsengine/testdata/simple/kubernetes_expected.json#L122-L204

Comment (raveeram) [reply examples/Templates/azuredeploy.json]: Updated.

File: examples/Templates/azuredeploy.parameters.json (hunk 1 above)
Comment (raveeram) [reply examples/Templates/azuredeploy.parameters.json]: done

File: examples/Templates/azuredeploy.parameters.json (hunk 2 above)
Comment (raveeram) [reply examples/Templates/azuredeploy.parameters.json]: done

**Review by anhowe — APPROVED** [review]
lgtm

**Pull request merged**